packages/next/src/server/lib/router-utils/setup-dev-bundler.ts

```diff
@@ -913,6 +916,52 @@ async function startWatcher(opts: SetupOpts) {
     return { finished: false }
   }
 
+  async function logError(
+    err: unknown,
+    type?: 'unhandledRejection' | 'uncaughtException' | 'warning' | 'app-dir'
+  ) {
+    if (err instanceof ModuleBuildError) {
+      // Errors that may come from issues from the user's code
+      Log.error(err.message)
+    } else if (err instanceof TurbopackInternalError) {
+      const levelMarker = '[FATAL]'
+      const fatalLogPath = path.join(opts.dir, '.next', 'fatal.log')
+      let logStat
+      try {
+        logStat = await stat(fatalLogPath)
+      } catch {}
+
+      if (logStat && logStat.size > 500 * 1024) {
+        // If the log file is greater than 500KB, truncate the least recent error
+        // to prevent the log from growing indefinitely
+        await writeFile(
+          fatalLogPath,
+          (await readFile(fatalLogPath, 'utf8'))
+            .split(levelMarker)
+            .slice(1)
+            .join(levelMarker)
+        )
+      }
```

Comment:
I'm 100% cool with this approach. Another potential option here:
- Create a unique `.next/fatal.log` file for each run, using either timestamp or pid to create a unique file name (`.next/fatal.12345.log`). In the rare case there's already a file the first time we try to write to it during the process's lifetime, just delete it.
- Cap the length of each individual `.next/fatal.log` file to 500 KB by silently skipping writes once it exceeds that length.